unison-cli/transcripts/Transcripts.hs

```diff
@@ -7,60 +7,56 @@ module Main (main) where
 
 import Data.Bifunctor (second)
 import Data.List
-import Data.Text
-  ( pack,
-    unpack,
-  )
+import qualified Data.Text as Text
 import EasyTest
-import Shellmet () -- import instance for IsString ([Text] -> IO())
 import System.Directory
 import System.Environment (getArgs)
 import System.FilePath
-  ( splitFileName,
-    takeBaseName,
+  ( replaceExtension,
+    splitFileName,
     takeExtensions,
     (</>),
   )
 import System.IO.CodePage (withCP65001)
-import System.Process (readProcessWithExitCode)
+import Unison.Codebase.Init (withTemporaryUcmCodebase)
+import qualified Unison.Codebase.SqliteCodebase as SC
+import Unison.Codebase.TranscriptParser (TranscriptError (..), withTranscriptRunner)
 import Unison.Prelude
 
 data TestConfig = TestConfig
   { matchPrefix :: Maybe String
   }
   deriving (Show)
 
-type TestBuilder = FilePath -> FilePath -> [String] -> String -> Test ()
+type TestBuilder = FilePath -> [String] -> String -> Test ()
 
 testBuilder ::
-  FilePath -> FilePath -> [String] -> String -> Test ()
-testBuilder ucm dir prelude transcript = scope transcript $ do
-  io $ fromString ucm args
+  FilePath -> [String] -> String -> Test ()
+testBuilder dir prelude transcript = scope transcript $ do
+  outputs <- io . withTemporaryUcmCodebase SC.init "transcript" $ \(codebasePath, codebase) -> do
```

Comment:
I experimented with different currying strategies to allow re-using the same transcript runner across all transcript tests, but it didn't have any perceivable performance difference so I just used the version that was clearest to read.